docs/reference/glossary.md

```diff
@@ -181,22 +187,58 @@ labeled using [tags](#tag).
 Here is an example of the shared [nginx repository](https://hub.docker.com/_/nginx/)
 and its [tags](https://hub.docker.com/r/library/nginx/tags/)
 
-## Swarm
 
-[Swarm](https://github.com/docker/swarm) is a native clustering tool for Docker.
-Swarm pools together several Docker hosts and exposes them as a single virtual
-Docker host. It serves the standard Docker API, so any tool that already works
-with Docker can now transparently scale up to multiple hosts.
+## service
+
+A [service](../swarm/how-swarm-mode-works/services.md) is the definition of how
+you want to run your application containers in a swarm. At the most basic level
+a service  defines which container image to run in the swarm and which commands
+to run in the container. For orchestration purposes, the service defines the
+"desired state", meaning how many containers to run as tasks and where to deploy
+the containers.
+
+Frequently a service is a microservice within the context of some larger
+application. Examples of services might include an HTTP server, a database, or
+any other type of executable program that you wish to run in a distributed
+environment.
+
+## service discovery
+
+Swarm mode [service discovery](../swarm/networking.md) is a DNS component
+internal to the swarm that automatically assigns each service in the swarm a VIP
+and DNS entry.
+
+## swarm
```

Comment:
I think we should create two entries
* swam
  A cluster of one or more Docker Engines running in swarm mode. You create a swarm with the `docker swarm init` command, and join nodes to the cluster with the `docker swarm join` command.

* Docker Swarm
A tool for creating and managing Docker Engine clusters. [Learn more about Docker Swarm](https://docs.docker.com/swarm/overview/).

Most of the functionality provided by Docker Swarm is now available natively in Docker 1.12, with the `docker swarm` command.